core/app/models/spree/promotion_code.rb

```diff
@@ -39,4 +40,8 @@ def usage_limit
   def downcase_value
     self.value = value.downcase
   end
+
+  def remove_excess_whitespace
+    value&.gsub! /\s+/, ''
```

Comment:
We already have a validation for code presence so I don't think we need to consider `nil` values.  Also I think we should use `self.value = value.strip` (not bang method and not worry about internal whitespace).  Open to others' thoughts on that.